client/controllers/events/eventFiltration.coffee

```diff
@@ -3,9 +3,9 @@ import regionToCountries from '/imports/regionToCountries.json'
 
 formatDateForInput = (date) ->
   unless date
-    return moment.utc().format('YYYY-MM-DD')
+    return moment()
```

Comment:
I was trying to figure out why there is a discrepancy when the dates are compared in the picker event.